libs/langgraph/langgraph/graph/schema_utils.py

```diff
@@ -139,11 +163,13 @@ def tuple_coercer(v: Any, d: Any) -> Any:
         if origin is Union:
             uargs = get_args(field_type)
             subs, none_in_union = [], False
-            for arg in uargs:
+            for ix, arg in enumerate(uargs):
                 if arg is type(None):
                     none_in_union = True
                 else:
-                    subs.append(self._build_coercer(arg))
+                    subs.append(
+                        self._build_coercer(arg, depth - 1, throw=ix < len(uargs) - 1)
```

Comment:
I'm still throwing if it's one of the initial values in the union check so we can check the next one in sequence.

Not sure a better way to handle this that aligns with pydantic's expectations